tck/src/main/java/org/eclipse/microprofile/concurrency/tck/cdi/CDIContextTest.java

```diff
@@ -0,0 +1,164 @@
+                .addClass(SessionScopedBean.class)
+                .addClass(ConversationScopeBean.class)
+                .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml");
+        System.out.println(war.toString(true));
+        return war;
+    }
+    
+    /**
+     * Set some state on Request, Session, and Conversation scoped beans and verify
+     * the state is propagated to the thread where the other task runs.
+     */
+    @Test
+    public void testCDIMECtxPropagate() throws Exception {
+        ManagedExecutor propagateCDI = ManagedExecutor.builder().build();
+        try {
+            checkCDIPropagation(true, "testCDI_ME_Ctx_Propagate-REQUEST", propagateCDI, requestBean);
+            checkCDIPropagation(true, "testCDI_ME_Ctx_Propagate-SESSION", propagateCDI, sessionBean);
+            checkCDIPropagation(true, "testCDI_ME_Ctx_Propagate-CONVERSATION", propagateCDI, conversationBean);
+        } 
+        finally {
+            propagateCDI.shutdown();
+        }
+    }
+
+    /**
+     * Set some state on Request, Session, and Conversation scoped beans and verify
+     * the state is cleared on the thread where the other task runs.
+     */
+    @Test
+    public void testCDIMECtxClear() throws Exception {
+        ManagedExecutor propagatedNone = ManagedExecutor.builder()
+                .propagated() // none
+                .cleared(ThreadContext.ALL_REMAINING)
+                .build();
+        try {
+            checkCDIPropagation(false, "testCDI_ME_Ctx_Clear-REQUEST", propagatedNone, requestBean);
+            checkCDIPropagation(false, "testCDI_ME_Ctx_Clear-SESSION", propagatedNone, sessionBean);
+            checkCDIPropagation(false, "testCDI_ME_Ctx_Clear-CONVERSATION", propagatedNone, conversationBean);
+        } 
+        finally {
+            propagatedNone.shutdown();
+        }
+    }
+
+    private void checkCDIPropagation(boolean expectPropagate, String stateToPropagate, ManagedExecutor me, AbstractBean bean) throws Exception {
+        bean.setState(stateToPropagate);
+        CompletableFuture<String> cf = me.supplyAsync(() -> {
+            String state = bean.getState();
+            System.out.println(stateToPropagate + " state=" + state);
+            return state;
+        });
+        assertEquals(cf.get(TIMEOUT_MIN, TimeUnit.MINUTES), expectPropagate ? stateToPropagate : AbstractBean.UNINITIALIZED);
+    }
+
+    /**
+     * Set some state on a request scoped bean, then verify a contextualized callable
+     * has the state propagated to it when ran on the same thread.
+     */
+    @Test
+    public void testCDITCCtxPropagate() throws Exception {
+        ThreadContext defaultTC = ThreadContext.builder().build();
+
+        requestBean.setState("testCDIContextPropagate-STATE2");
+        Callable<String> getState = defaultTC.contextualCallable(() -> {
+            String state = requestBean.getState();
+            System.out.println("testCDIContextPropagate#2 state=" + state);
```

Comment:
Probably testing leftovers? There are multiple occurrences throughout the test, please remove.

Reply:
I was intending to leave in place because it can provide some useful debug information, (especially when impls flow System.out through a logger that includes timestamp and thread ID) in the event of a test failure. I can remove them though